src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp

```diff
@@ -0,0 +1,77 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "ApparentHorizons/TagsDeclarations.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarTensor {
+
+/// @{
+/*!
+ * \brief The scalar charge per unit solid angle.
+ *
+ * \details This function calculates the integrand of:
+ * \f{align*}
+ * q = - \dfrac{1}{4 \pi} \oint dA \Phi_i n^{i},
+ * \f}
+ * where \f$ n^{i} \f$ is the unit (outward) normal of the surface.
+ *
+ * For a spherically symmetric scalar, this value will coincide with the value
+ * as extracted from the $r^{-1}$ decay in
+ * \f[ \Psi \sim \phi_\infty + q / r + \cdots~. \f]
+ */
+template <typename Frame>
+void scalar_charge_integrand(
+    const gsl::not_null<Scalar<DataVector>*> result,
+    const tnsr::i<DataVector, 3, Frame>& phi,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector);
+/// @}
+
+} // namespace ScalarTensor
+
+namespace ScalarTensor::StrahlkorperScalar::Tags {
+
+/*!
+ * \brief The scalar charge per unit area.
+ *
+ * \details This tag holds the integrand of:
+ * \f{align*}
+ * q = - \dfrac{1}{4 \pi} \oint dA \Phi_i n^{i},
+ * \f}
+ * where \f$ n^{i} \f$ is the unit (outward) normal of the surface.
+ *
+ * For a spherically symmetric scalar, this value will coincide with the value
+ * as extracted from the $r^{-1}$ decay in
+ * \f[ \Psi \sim \phi_\infty + q / r + \cdots~. \f]
+ */
+struct ScalarChargeIntegrand : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+/*!
+ * \brief Compute tag for the scalar charge per unit area.
+ *
+ * \see ScalarTensor::StrahlkorperScalar::Tags::ScalarChargeIntegrand
+ */
+template <typename Frame>
+struct ScalarChargeIntegrandCompute : ScalarChargeIntegrand, db::ComputeTag {
+  static constexpr size_t Dim = 3;
+  using base = ScalarChargeIntegrand;
+  static constexpr auto function =
+      static_cast<void (*)(gsl::not_null<Scalar<DataVector>*>,
+                           const tnsr::i<DataVector, Dim, Frame>&,
+                           const tnsr::I<DataVector, Dim, Frame>&)>(
+          &ScalarTensor::scalar_charge_integrand<Frame>);
```

Comment:
you don't even need the `static_cast` since you have only a single overload.

Reply:
Okay, I have removed the `static_cast`